tests/e2e/coverage/controller-manager.yaml

```diff
@@ -0,0 +1,49 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: numaplane-controller-manager
+  namespace: numaplane-system
+  labels:
+    app.kubernetes.io/name: controller-manager
+    app.kubernetes.io/component: controller-manager
+    app.kubernetes.io/part-of: numaplane
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: controller-manager
+      app.kubernetes.io/part-of: numaplane
+      app.kubernetes.io/component: controller-manager
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-container: manager
+      labels:
+        app.kubernetes.io/name: controller-manager
+        app.kubernetes.io/part-of: numaplane
+        app.kubernetes.io/component: controller-manager
+    spec:
+      containers:
+        - image: quay.io/numaproj/numaplane-controller:latest
```

Comment:
I wonder if we want a distinct image name or image tag for this one, to avoid confusion